scalafmt-core/shared/src/main/scala/org/scalafmt/config/SortSettings.scala

```diff
@@ -0,0 +1,53 @@
+package org.scalafmt.config
+
+import metaconfig._
+
+@DeriveConfDecoder
+case class SortSettings(
+    order: Vector[SortSettings.ModKey]
+)
+
+object SortSettings {
+
+  implicit val SortSettingsModKeyReader: ConfDecoder[ModKey] =
+    ReaderUtil.oneOfIgnoreBackticks[ModKey](
+      `implicit`,
+      `final`,
+      `sealed`,
+      `abstract`,
+      `override`,
+      `private`,
+      `protected`,
+      `lazy`
+    )
+
+  val defaultOrder: Vector[ModKey] =
+    Vector(
+      `implicit`,
+      //
+      `final`,
+      `sealed`,
+      `abstract`,
+      //
+      `override`,
+      //
+      `private`,
+      `protected`,
+      //
+      `lazy`
+    )
+
+  def default: SortSettings =
+    SortSettings(defaultOrder)
+
+  sealed trait ModKey
```

Comment:
`extends Product` allows you to use `.productPrefix` from the docs instead of the `getClass` hack